core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -615,6 +618,132 @@ private function addBackupNode(): NodeDefinition
+                ->addDefaultsIfNotSet()
+                ->children()
+                    ->scalarNode('adapter')
+                        ->info('The Flysystem adapter that should be used, e.g. "local" or "aws"')
+                        ->defaultValue('local')
+                        ->isRequired()
+                        ->cannotBeEmpty()
+                    ->end()
+                    ->arrayNode('options')
+                        ->info('Adapter options as key-value-pairs, e.g. for the "local" adapter, set "directory" to the storage\'s root path')
+                        ->variablePrototype()
+                        ->end()
+                        ->defaultValue([])
+                    ->end()
+                    ->scalarNode('visibility')
+                        ->defaultNull()
+                    ->end()
+                    ->booleanNode('case_sensitive')
+                        ->defaultTrue()
+                    ->end()
+                    ->booleanNode('disable_asserts')
+                        ->defaultFalse()
+                    ->end()
+                ->end()
+            ;
+        };
+
+        $dbafsSection = static function (): NodeDefinition {
+            return (new TreeBuilder('dbafs'))
+                ->getRootNode()
+                ->info('DBAFS configuration, automatically enabled when configured.')
+                ->canBeEnabled()
+                ->children()
+                    ->scalarNode('table')
+                        ->info('The database table to use for synchronizing records. Defaults to "tl_<bucket_name>"')
+                        ->cannotBeEmpty()
+                        ->defaultValue('')
+                    ->end()
+                    ->enumNode('hash_algorithm')
+                        ->info('The hash algorithm to use for comparing file contents.')
+                        ->cannotBeEmpty()
+                        ->values(hash_algos())
+                        ->defaultValue('md5')
+                    ->end()
+                    ->integerNode('max_file_size')
+                        ->info('Files with a larger size than configured, won\'t be considered when synchronizing. Set the maximum in bytes - you can also use a magnitude postfix like "400M" or "2Gi".')
+                        ->min(1)
+                        ->defaultValue(2147483648) // 2Gi
+                        ->beforeNormalization()
+                            ->ifString()
+                            ->then(
+                                static fn (string $value): int => (int) (new NumberComparator($value))->getTarget()
+                            )
+                        ->end()
+                    ->end()
+                    ->integerNode('bulk_insert_size')
+                        ->info('We are using bulk inserts to improve the database performance when writing lots of entries. You can fine tune the size here.')
+                        ->min(1)
+                        ->defaultValue(100)
```

Comment:
Random thought: Would it make sense to have something like PHP's OPCache `validate_tstamps` setting here to check for the last modified date of the file when comparing the hashes.
Just because I want to omit downloading a 4 GB file from S3 to calculate the hash if in fact we knew that it's still going to be the same if the last modified date did not change. Should be enabled by default in order to increase performance and decrease traffic usage but one should be able to disable it if they know that a given file system does not update the last modified date for example. Wdyt about this idea?

Reply:
Interesting, I like this idea!

I think we could implement it in a future PR without breaking anything and keep this one as big as it already is: :smile: 
* introduce a new fileld `lastModified` to `tl_files`
* add the function to the `Dbafs` class (disabled by default, also makes sense, because the service should IMO require the minimal amount of columns by default)
* enable it in the config

Reply:
True, the new abstraction would allow for something like that. We should definitely not forget about that. But creating an issue for a PR that's not even merged yet sounds a bit wrong :D

Reply:
see bf7c0b07aa61983aa05d5b1fb4f6afa837467e3f - it must be opt in, I guess (even when implemented into the `Dbafs`) because the only sensible thing then is to include the file time in the hashes as well, which would force people to recompute them.

Reply:
Ok it's implemented now. :see_no_evil: